### PR TITLE
DB-9139 Print reason to choose engine in explain

### PR DESCRIPTION
**Pull request by arnaud-splice**

### CODE REVIEW

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST FAILED

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh5.12.2

File: db-engine/src/main/java/com/splicemachine/db/impl/ast/PlanPrinter.java [hunk 1]
```diff
@@ -93,10 +93,10 @@ public Visitable defaultVisit(Visitable node) throws StandardException {
             m.put(query,orderedNodes);
             if (LOG.isDebugEnabled()){
                 DataSetProcessorType currentType = rsn.getLanguageConnectionContext().getDataSetProcessorType();
-                currentType = currentType.combine(queryHintedForcedType(orderedNodes));
-                boolean useSpark = currentType.isSpark() || currentType.isDefaultControl() && PlanPrinter.shouldUseSpark(orderedNodes, true);
+                currentType = currentType.combine(rsn.getCompilerContext().getDataSetProcessorType());
+                currentType = currentType.combine(determineSpark(orderedNodes));
```
Comment (yxia92) [inline db-engine/src/main/java/com/splicemachine/db/impl/ast/PlanPrinter.java]: Do we really this line? I think the dataSetProcessorType obtained from rsn.getCompilerContext().getDataSetProcessorType() is the final decision, and also the one that is passed to activation, so we don't need to go through every node in orderedNodes.

**Comment by spliceci** [comment]
TEST FAILED

File: db-engine/src/main/java/com/splicemachine/db/impl/ast/PlanPrinter.java (hunk 1 above)
Comment (arnaud-splice) [reply db-engine/src/main/java/com/splicemachine/db/impl/ast/PlanPrinter.java]: You're right, I merely refactored that function while keeping the behaviour intact, but we don't need that anymore. We also don't need the languagecontext.type as this one was put in compilercontext.type during the optimisation phase.

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**